graylog2-web-interface/src/views/logic/SearchResult.js

```diff
@@ -1,18 +1,43 @@
-import _ from 'lodash';
+import { mapValues, get, compact } from 'lodash';
 import QueryResult from './QueryResult';
 import SearchError from './SearchError';
 
 class SearchResult {
   constructor(result) {
     this.result = result;
 
-    this.results = _.mapValues(result.results, queryResult => new QueryResult(queryResult));
-    this.errors = _.get(result, 'errors', []).map(error => new SearchError(error));
+    this.results = mapValues(result.results, queryResult => new QueryResult(queryResult));
+    this.errors = get(result, 'errors', []).map(error => new SearchError(error));
   }
 
   forId(queryId) {
     return this.results[queryId];
   }
+
+  updateSearchTypes(searchTypeResults) {
+    const updatedResult = new SearchResult(this.result).result;
```

Comment:
In a previous version https://github.com/Graylog2/graylog2-server/pull/6903/commits/04fce92df75f52884f94d9efa92694d645638ab0
we just used `this.result` for `updatedResult`.
Because we are adjusting the object in line 21, the old implementation would mutate the "current version" of this.result, right?
It feels a bit strange to run `new SearchResult` twice.